ReSwift/ReSwift-Thunk
# Implementation for iOS

### This PR adds
- Code of the framework and the tests
- iOS target (following the naming conventions in ReSwift)
- iOS shared scheme 
- iOS test target
- Cartfile (I'm not sure if we should include `Cartfile.resolved`, but it's not included in `.gitignore`)

Let me know if you like the setup and the naming, I didn't' want to create the other targets before you can review everything. 
### How to test
- Configure a new iOS project with a this line `github "ReSwift/ReSwift-Thunk" "implementation"` in its Carfile, and follow Carthage instructions to integrate both ReSwift and ReSwift-Thunk to the project.
- Use the framework importing it: `import ReSwift_Thunk`.
- Setup a store with a thunks middleware and dispatch some thunks to see if it works.
### Next
- Set up the rest of the targets
- Add Cocoapods support
- Add SPM support
